### DISCUSSION TITLE
ImgViewerAnimatedGif fails included GIF file

### DISCUSSION THREAD

**Discussion by timmgleason** [moononournation/Arduino_GFX#100] [Q&A] [answered]
Working on displaying a GIF on a small round LCD (G9A01 IPS) with a Seeediuno with an SD card attached.
I've put the included GIF files onto the SD card, along with a text file, but the **gd_GIF *gif = gifClass.gd_open_gif(&gifFile);** test fails the file every time. I've tried other GIF files as well. Checked what the class is doing and indeed verified that the first line of the GIF is "**GIF89a**, which is what the class appears to be looking for.
I have verified that I am actually reading from the SD card by changing the serial print output message to the text stored in the text file from the SD card.
I've tried every thing that I can think of here.

**NOTE - I am **not** using the SEEED_FS.h & SEEED_SD.h as I couldn't get them to work opening the SD card. I am just using the standard SD.h library.

Thanks!!

**Comment by moononournation** [comment]
What dev board are you using?

**Comment by timmgleason** [comment, marked as answer]
I am using a Seeeduino XIAO. But as I stated, I am not using the Seeed_Arduino_FS libraries, just the standard SD libraries.
I couldn't get the Seeed specific ones working, but the standard ones I could.

**Reply by moononournation** [reply]
XIAO only have 1 SPI, you need to share the SPI with display.

**Reply by timmgleason** [reply]
I'll give that a try!

**Reply by timmgleason** [reply]
I am currently sharing the CS pin between the 2.
Is that the SPI pin you are referring to? (I'm kind of new to Arduino tinkering)

**Reply by timmgleason** [reply]
Also the MOSI pin

**Reply by moononournation** [reply]
you may refer to this code how to share SPI between SD and display: https://github.com/moononournation/RGB565_video/blob/master/SD_GIF_video/SD_GIF_video.ino

**Reply by timmgleason** [reply]
Well, I have looked through that sketch and made a few changes, mainly involving getting the Seeed libraries working properly, but now it is hanging up on setting the *buf.

> 12:37:38.021 -> Initializing display
> 12:37:38.955 -> new_table() malloc failed: 32776
> 12:37:38.955 -> buf malloc failed!
I've tried both the routine in the original file and the SD_GIF_video.ino file you pointed me to.
I have tried setting the value manually as well.
I actually expected that number to be higher, as the GIF file is 240x240 (57600). I tried using the smaller GIF (64x64) and got the same error. I actually get the same number no matter what the value is.

I would set a static value, knowing that my display is only 240x240 maximum, but I must confess to being ignorant of the math happening in the assignment line, _uint8_t *buf = (uint8_t *)malloc(gif->width * gif->height);_.

Cheers

**Reply by timmgleason** [reply]
UPDATE:
If I use the smaller provided GIF and set the number to 4096 (_uint8_t *buf = (uint8_t *)malloc(4096);_ or _uint8_t *buf = (uint8_t *)malloc(gif->width * gif->height);_), I still see the malloc error, but it "starts" the GIF playing.

> 13:25:45.754 -> Initializing display
> 13:25:46.734 -> new_table() malloc failed: 32776
> 13:25:46.734 -> GIF video start

I do not actually get the image displayed, only some half-tone lines and a white bar at the top of the display.

Progress, but still not working proper.

**Reply by moononournation** [reply]
XIAO only have 32k memory, so it cannot allocate 240x240=57.6k memory

**Reply by timmgleason** [reply]
Yes, this is what I was afraid of. More flash RAM than the Nano, but same SRAM.
Apparently not even enough for the smaller GIF either, 64x64=4096.
I will try it with the Nano Connect RP2040, 16MB Flash, 520K SRAM, also does away with my power control issue since that is built into the Nano architecture already.

Cheers!
